src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -535,9 +547,19 @@ ss::future<> partition_downloader::download_segment_file(
         co_await ss::remove_file(localpath.string());
     }
 
-    auto stream = [this, part, remote_path, _localpath{localpath}, _otl{otl}](
+    model::offset min_offset;
+    model::offset max_offset;
+    auto stream = [this,
+                   _part{part},
+                   _remote_path{remote_path},
+                   _localpath{localpath},
+                   _otl{otl},
+                   &min_offset,
+                   &max_offset](
                     uint64_t len,
                     ss::input_stream<char> in) -> ss::future<uint64_t> {
+        auto part{_part};
+        auto remote_path{_remote_path};
```

Comment:
So I think in this lambda `remote_path` and `part` are actually okay - they are used only before the first `co_await`. But `min_offset` and `max_offset` are not - they must be saved on the coroutine stack.